### ISSUE TITLE
Setting root passwd via user.present gives invalid results

### ISSUE THREAD

**Issue by seanjnkns** [saltstack/salt#30208]
Using a simple state such as:
```sls
root:
  user.present:
    - password: '$6$xcRjzm3.MXHaqzUd$t0sOmNQLuzg0GfLfDE3Sb7XL9vNhdzSFq8L.OtzaIVlcazJUj4D7B3MrDBwqG6aUMbX0SLbYLjYlxEojX2S.K0'
```
Will give the following results:
```yaml
local:
----------
          ID: root
    Function: user.present
      Result: False
     Comment: These values could not be changed: {'passwd': '$6$xcRjzm3.MXHaqzUd$t0sOmNQLuzg0GfLfDE3Sb7XL9vNhdzSFq8L.OtzaIVlcazJUj4D7B3MrDBwqG6aUMbX0SLbYLjYlxEojX2S.K0'}
     Started: 11:16:23.509460
    Duration: 2.775 ms
     Changes:
```
However, checking /etc/shadow, the password was changed.  Subsequent runs keeps throwing the same results even though the password didn't need to change, nor did.  Seems to be a false positive.

Tested on 2015.8.1 and 2015.8.3

**Comment by jfindlay** [comment]
Thanks for reporting, @seanjnkns.

**Comment by seanjnkns** [comment]
After testing this numerous times over between 2015.8.1, 2015.8.3 and development branch, using the exact same state, I can't retrigger the problem.  Not sure if we want to keep this open or not, but if so, I'll see if I can retrigger the issue again and retrace my steps.

**Comment by jfindlay** [comment]
@seanjnkns, thanks for your work on this.  Let's close it for now unless you're able to reproduce it again so you don't have to worry about it if you'd rather move on.

**Comment by c-wicklein** [comment]
I encountered this symptom with a minion running Ubuntu 14.04.  I don't know what the state of the machine was before running salt, but I wasn't able to change the password for some non-root user with the same error message.  I noticed that the user's last password change field in /etc/shadow was set to -1.  Editing /etc/shadow and setting this to a positive integer fixed the problem.  It's possible that @seanjnkns accidentally discovered the same fix by resetting root's password (e.g. with passwd(1).)
